engine/buildkit/filesync.go

```diff
@@ -110,54 +107,42 @@ func (c *Client) LocalDirExport(
 	}
 	defer cancel(errors.New("local dir export done"))
 
-	destPath = path.Clean(destPath)
-
-	res, err := c.Solve(ctx, bkgw.SolveRequest{Definition: def})
-	if err != nil {
-		return fmt.Errorf("failed to solve for local export: %w", err)
-	}
-	cacheRes, err := ConvertToWorkerCacheResult(ctx, res)
-	if err != nil {
-		return fmt.Errorf("failed to convert result: %w", err)
-	}
-
-	// TODO: lift this to dagger
-	exporter, err := c.Worker.Exporter(bkclient.ExporterLocal, c.SessionManager)
+	outputFS, err := fsutil.NewFS(srcPath)
 	if err != nil {
 		return err
 	}
 
-	expResult, err := exporter.Resolve(ctx, 0, nil)
+	clientMetadata, err := engine.ClientMetadataFromContext(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to resolve exporter: %w", err)
+		return err
 	}
 
-	clientMetadata, err := engine.ClientMetadataFromContext(ctx)
+	caller, err := c.SessionManager.Get(ctx, clientMetadata.ClientID, false)
 	if err != nil {
-		return fmt.Errorf("failed to get requester session ID: %w", err)
+		return err
 	}
 
+	destPath = path.Clean(destPath)
 	ctx = engine.LocalExportOpts{
 		Path:        destPath,
 		Merge:       merge,
 		RemovePaths: removePaths,
 	}.AppendToOutgoingContext(ctx)
 
-	_, descRef, err := expResult.Export(ctx, cacheRes, nil, clientMetadata.ClientID)
-	if err != nil {
-		return fmt.Errorf("failed to export: %w", err)
-	}
-	if descRef != nil {
-		descRef.Release()
+	lbl := "copying files"
+	progress := local.NewProgressHandler(ctx, lbl)
```

Comment:
nit: do we need this? I've been seeing buildkit progress stuff pop in pprof cpu profiles occasionally and was thinking about tearing it out sometime since we just use our own telemetry.

not a blocker if it's somehow load bearing still for now

Reply:
good catch -- fsutils will avoid calling it if it's set to nil, it seems safe to remove.